app/controllers/RestController.scala

```diff
@@ -73,7 +75,26 @@ class RestController @Inject()(
 
     def onReset: Action[AnyContent] = silhouette.SecuredAction.async {
         implicit request: SecuredRequest[DefaultEnv, AnyContent] =>
-            serverCtrl.reset()
+            execute(serverCtrl.reset)
             Future.successful(NoContent)
     }
+
+    def execute(fn: => Unit): Unit = {
+        try {
+            fn
+        } catch {
+            case e: ControllerException =>
+                sendJsonToWebSocket(createErrorJson(e.message))
+        }
+    }
+
+    def sendJsonToWebSocket(json: JsValue): Unit = {
```

Comment:
is it necessary to have this in the `HomeController` and `RestController`? if we need it here we could just make the def in the `HomeController` public